apps/sms/views/shared/js/utils.js

```diff
@@ -945,24 +945,33 @@
     },
 
     /**
-     * Returns Promise instance that is resolved instantly if document is
-     * visible (document.hidden === false), othewise it's resolved only once
-     * `visibilitychange` event is fired.
-     * @returns {Promise.<void>}
+     * Returns Promise instance that is resolved only once specified event is
+     * fired on the specified event target.
+     * @param {EventTarget} eventTarget Event target to listen event on.
+     * @param {String} eventName Name of the event to listen for.
+     * @returns {Promise.<*>} Promise resolved with event object if any.
```

Comment:
Yeah, it's supposed to work only with [EventTarget](https://developer.mozilla.org/en/docs/Web/API/EventTarget) compatible WebAPI - will replace ```*``` with ```Event```, thanks!